src/web/tools/PointerEventManager.ts

```diff
@@ -115,7 +119,28 @@ export default class PointerEventManager extends EventManager {
       this.onPointerCancel(adaptedEvent);
       this.markAsOutOfBounds(adaptedEvent.pointerId);
       this.activePointersCounter = 0;
+      this.trackedPointers.clear();
     });
+
+    this.view.addEventListener(
+      'lostpointercapture',
+      (event: PointerEvent): void => {
+        const adaptedEvent: AdaptedEvent = this.mapEvent(event, EventTypes.UP);
+
+        if (this.trackedPointers.has(adaptedEvent.pointerId)) {
+          // in some cases the `pointerup` event is not fired, but `lostpointercapture` is
+          // we simulate the `pointerup` event here to make sure the gesture handler stops tracking it
+          this.trackedPointers.delete(adaptedEvent.pointerId);
+
+          if (--this.activePointersCounter > 0) {
+            adaptedEvent.eventType = EventTypes.ADDITIONAL_POINTER_UP;
```

Comment:
Shouldn't we treat this event similarly to how we process CANCEL on Android and iOS? The issue with this approach is that sending UP event may result in some action being triggered (e.g. click). I think what could happen here is that the pointer may be lost due to the browser window changing focus, or some alert being presented?

Reply:
Makes sense to cancel the gesture in that case, I've updated the PR (also I've unified how `pointercancel` event is handled by the handlers).

> I think what could happen here is that the pointer may be lost due to the browser window changing focus, or some alert being presented?

In this specific case, it was happening when the pointer was moving really fast. The event stream would end without receiving either `up` or `cancel` events.